src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/TestsConnectionFactory.cs

```diff
@@ -1,26 +1,41 @@
 ﻿using Microsoft.Data.Sqlite;
 using Microsoft.EntityFrameworkCore;
+using SFA.DAS.ApprenticeCommitments.Data.Models;
 using SFA.DAS.ApprenticeCommitments.Infrastructure;
+using System;
 using System.Data.Common;
 
 namespace SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests
 {
+    public interface ITestsDbConnectionFactory : IConnectionFactory
+    {
+        string ConnectionString { get; }
+
+        void EnsureCreated(ApprenticeCommitmentsDbContext dbContext);
+
+        void EnsureDeleted(ApprenticeCommitmentsDbContext dbContext);
+    }
+
     public class TestsDbConnectionFactory : IConnectionFactory
     {
-        private static IConnectionFactory factory { get; }
-        public static string ConnectionString { get; }// => $"Data Source={AcceptanceTestsData.AcceptanceTestsDatabaseName}";
+        private static ITestsDbConnectionFactory factory;
+        public static string ConnectionString => factory.ConnectionString;
 
         static TestsDbConnectionFactory()
         {
             //*
             factory = new SqliteTestsConnectionFactory();
-            ConnectionString = $"Data Source={AcceptanceTestsData.AcceptanceTestsDatabaseName}";
             /*/
             factory = new SqlServerTestsConnectionFactory();
-            ConnectionString = "Data Source=.;Initial Catalog=SFA.DAS.ApprenticeCommitments.AcceptanceTests;Integrated Security=True;";
             //*/
         }
 
+        internal void EnsureCreated(ApprenticeCommitmentsDbContext dbContext)
+            => factory.EnsureCreated(dbContext);
```

Comment:
Should this is doing 
EnsureDeleted()
EnsureCreated()

as below

Reply:
I think it depends on the database provider.  `SqliteTestsConnectionFactory` does in fact do exactly that in `EnsureCreated`.  `SqlServerTestsConnectionFactory` needs to check the existence of the db before it can truncate tables.

There's probably some room for improvement next time we look at it.